server/src/main/java/org/apache/cloudstack/consoleproxy/ConsoleAccessManagerImpl.java

```diff
@@ -0,0 +1,489 @@
+import com.cloud.host.HostVO;
+import com.cloud.hypervisor.Hypervisor;
+import com.cloud.resource.ResourceState;
+import com.cloud.server.ManagementServer;
+import com.cloud.servlet.ConsoleProxyClientParam;
+import com.cloud.servlet.ConsoleProxyPasswordBasedEncryptor;
+import com.cloud.storage.GuestOSVO;
+import com.cloud.user.Account;
+import com.cloud.user.AccountManager;
+import com.cloud.utils.Pair;
+import com.cloud.utils.Ternary;
+import com.cloud.utils.component.ManagerBase;
+import com.cloud.utils.db.EntityManager;
+import com.cloud.utils.exception.CloudRuntimeException;
+import com.cloud.vm.UserVmDetailVO;
+import com.cloud.vm.VirtualMachine;
+import com.cloud.vm.VirtualMachineManager;
+import com.cloud.vm.VmDetailConstants;
+import com.cloud.vm.dao.UserVmDetailsDao;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.apache.cloudstack.api.command.user.consoleproxy.ConsoleEndpoint;
+import org.apache.cloudstack.context.CallContext;
+import org.apache.cloudstack.framework.config.ConfigKey;
+import org.apache.cloudstack.framework.security.keys.KeysManager;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import javax.inject.Inject;
+import javax.naming.ConfigurationException;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+public class ConsoleAccessManagerImpl extends ManagerBase implements ConsoleAccessManager {
+
+    @Inject
+    private AccountManager accountManager;
+    @Inject
+    private VirtualMachineManager virtualMachineManager;
+    @Inject
+    private ManagementServer managementServer;
+    @Inject
+    private EntityManager entityManager;
+    @Inject
+    private UserVmDetailsDao userVmDetailsDao;
+    @Inject
+    private KeysManager keysManager;
+    @Inject
+    private AgentManager agentManager;
+    @Inject
+    private ConsoleProxyManager consoleProxyManager;
+
+    private static KeysManager secretKeysManager;
+    private final Gson gson = new GsonBuilder().create();
+
+    public static final Logger s_logger = Logger.getLogger(ConsoleAccessManagerImpl.class.getName());
+
+    private static Set<String> allowedSessions;
+
+    @Override
+    public boolean configure(String name, Map<String, Object> params) throws ConfigurationException {
+        ConsoleAccessManagerImpl.secretKeysManager = keysManager;
+        ConsoleAccessManagerImpl.allowedSessions = new HashSet<>();
+        return super.configure(name, params);
+    }
+
+    @Override
+    public ConsoleEndpoint generateConsoleEndpoint(Long vmId, String extraSecurityToken, String clientAddress) {
+        try {
+            if (accountManager == null || virtualMachineManager == null || managementServer == null) {
```

Comment:
We could use `org.apache.commons.lang3.ObjectUtils#anyNull()` here:

```suggestion
            if (ObjectUtils.anyNull(accountManager, virtualMachineManager, managementServer)) {
```